From c38999474b2759d63f39fd17bc77d4f78d09a95d Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:09 -0400
Subject: [tools] perf/timechart: Move wake_events list to 'struct timechart'

Message-id: <1408953210-26343-62-git-send-email-jolsa@redhat.com>
Patchwork-id: 88784
O-Subject: [PATCH RHEL7.1 BZ1133083 061/322] perf timechart: Move wake_events list to 'struct timechart'
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 436b0da02fc12970f2f5ecdef03160c7a401b121
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Nov 28 13:26:33 2013 -0300

description
===========
Removing another global variable.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-timechart.c b/tools/perf/builtin-timechart.c
index 8ee0ff1..0bda620 100644
--- a/tools/perf/builtin-timechart.c
+++ b/tools/perf/builtin-timechart.c
@@ -43,11 +43,13 @@
 
 struct per_pid;
 struct power_event;
+struct wake_event;
 
 struct timechart {
  struct perf_tool tool;
  struct per_pid  *all_data;
  struct power_event *power_events;
+ struct wake_event *wake_events;
  int   proc_num;
  unsigned int  numcpus;
  u64   min_freq, /* Lowest CPU frequency seen */
@@ -148,8 +150,6 @@ struct wake_event {
  const char *backtrace;
 };
 
-static struct wake_event     *wake_events;
-
 struct process_filter {
  char   *name;
  int   pid;
@@ -383,8 +383,8 @@ static void sched_wakeup(struct timechart *tchart, int cpu, u64 timestamp,
   we->waker = -1;
 
  we->wakee = wakee;
- we->next = wake_events;
- wake_events = we;
+ we->next = tchart->wake_events;
+ tchart->wake_events = we;
  p = find_create_pid(tchart, we->wakee);
 
  if (p && p->current && p->current->state == TYPE_NONE) {
@@ -764,7 +764,7 @@ static void draw_wakeups(struct timechart *tchart)
  struct per_pid *p;
  struct per_pidcomm *c;
 
- we = wake_events;
+ we = tchart->wake_events;
  while (we) {
   int from = 0, to = 0;
   char *task_from = NULL, *task_to = NULL;
-- 
1.7.1